src/OpalCompiler-Core/Behavior.extension.st

```diff
@@ -15,6 +15,26 @@ Behavior >> compile: code [
 	^self compile: code notifying: nil
 ]
 
+{ #category : '*OpalCompiler-Core' }
+Behavior >> compile: selector from: oldClass [
+	"Compile the method associated with selector in the receiver's method dictionary."
+
+	| method newMethod |
+	method := oldClass compiledMethodAt: selector.
+	newMethod := oldClass compiler
+		             source: (oldClass sourceCodeAt: selector);
```

Comment:
oldClass SourceCodeAt: selector -> method

Reply:
`oldClass SourceCodeAt: selector` returns the source code while #method contains the compiled method so it is not the same.

Or we could use `method sourceCode`